<a href="https://colab.research.google.com/github/francorte/cafa6-protein-function-prediction/blob/main/An%C3%A1lisis_nutricional_de_alimentos_de_comida_r%C3%A1pida.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# IMPORTANT: RUN THIS CELL IN ORDER TO IMPORT YOUR KAGGLE DATA SOURCES,
# THEN FEEL FREE TO DELETE THIS CELL.
# NOTE: THIS NOTEBOOK ENVIRONMENT DIFFERS FROM KAGGLE'S PYTHON
# ENVIRONMENT SO THERE MAY BE MISSING LIBRARIES USED BY YOUR
# NOTEBOOK.
import kagglehub
joebeachcapital_fast_food_path = kagglehub.dataset_download('joebeachcapital/fast-food')

print('Data source import complete.')


100%|██████████| 40.3k/40.3k [00:00<00:00, 12.8MB/s]

Extracting files...
Data source import complete.


# Análisis nutricional de alimentos de comida rápida

**Ciclo Formativo:** Grado Superior en Dietética y Nutrición  
**Módulo profesional:** Educación sanitaria / Control alimentario  
**Tipo de recurso:** Cuaderno docente con análisis de datos reales  
**Autor:** Francisco de la Corte  
**Curso académico:** 2024–2025

---

Este material está diseñado como **recurso docente** para el trabajo en aula
en Formación Profesional, con el objetivo de:

- desarrollar la capacidad de **interpretar datos nutricionales reales**,
- aplicar criterios profesionales en el análisis de alimentos,
- y **justificar recomendaciones dietéticas basadas en evidencias**.

El enfoque combina **nutrición, pensamiento crítico y análisis de datos**,
competencias clave en el perfil profesional del Técnico Superior en Dietética.

# Análisis nutricional de alimentos de comida rápida (FP Dietética y Nutrición)

## Objetivo del notebook
En este notebook vamos a trabajar con un dataset de productos de **comida rápida** para aprender a:

- Leer e interpretar **tablas nutricionales reales** (como las de un etiquetado o una ficha de producto).
- Identificar qué nutrientes destacan en estos alimentos (energía, grasas, azúcares, proteínas y sodio).
- Comparar productos entre sí y **argumentar** si serían adecuados o no para un consumo frecuente.
- Conectar el análisis con contenidos del ciclo: **educación alimentaria** y relación con el concepto de **ultraprocesado**.

## ¿Por qué usamos comida rápida?
Porque son alimentos:
- Muy conocidos por el alumnado (alto interés y cercanía).
- Con composiciones nutricionales muy variables.
- Útiles para entrenar el razonamiento: no se trata de “demonizar”, sino de **analizar datos y justificar conclusiones**.

# Descripción del dataset

Este dataset recoge información nutricional de distintos productos de comida rápida.  
**Cada fila representa un producto** (por ejemplo, un tipo de hamburguesa, una bebida, un postre, etc.).

Vamos a centrarnos en variables habituales en nutrición:

- **Calories (kcal):** energía total del producto.
- **Total Fat (g):** grasa total.
- **Saturated Fat (g):** grasa saturada (importante por su relación con salud cardiovascular).
- **Carbohydrates (g):** hidratos de carbono totales.
- **Sugars (g):** azúcares (parte de los hidratos, relevantes en bebidas y postres).
- **Protein (g):** proteínas.
- **Sodium (mg):** sodio (relacionado con exceso de sal y salud cardiovascular).

> Importante: en este notebook no buscamos “la respuesta única correcta”, sino practicar:
> - lectura de datos
> - comparación
> - y justificación nutricional con criterio.

In [ ]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("joebeachcapital/fast-food")

print("Path to dataset files:", path)

Using Colab cache for faster access to the 'fast-food' dataset.
Path to dataset files: /kaggle/input/fast-food


In [ ]:
import os

# Listar los archivos disponibles en el dataset
for file in os.listdir("/kaggle/input/fast-food"):
    print(file)

FastFoodNutritionMenuV3.csv
.nfs0000000068d86dbe000003c7
FastFoodNutritionMenuV2.csv


In [ ]:
import pandas as pd

# Cargar el dataset (versión más reciente)
df = pd.read_csv("/kaggle/input/fast-food/FastFoodNutritionMenuV3.csv")

# Vista rápida de los datos
df.head()

Company                           Item Calories Calories from\nFat  \
0  McDonald’s                      Hamburger      250                 80   
1  McDonald’s                   Cheeseburger      300                110   
2  McDonald’s            Double Cheeseburger      440                210   
3  McDonald’s                       McDouble      390                170   
4  McDonald’s   Quarter Pounder® with Cheese      510                230   

  Total Fat\n(g) Saturated Fat\n(g) Trans Fat\n(g) Cholesterol\n(mg)  \
0              9                3.5            0.5                25   
1             12                  6            0.5                40   
2             23                 11            1.5                80   
3             19                  8              1                65   
4             26                 12            1.5                90   

  Sodium \n(mg) Carbs\n(g) Fiber\n(g) Sugars\n(g) Protein\n(g)  \
0           520         31          2           6           12   
1           750         33          2           6           15   
2          1150         34          2           7           25   
3           920         33          2           7           22   
4          1190         40          3           9           29   

  Weight Watchers\nPnts  
0                 247.5  
1                   297  
2                   433  
3                   383  
4                   502

## Comprendiendo la estructura del dataset

Antes de analizar los alimentos, es fundamental entender **qué información contiene la tabla**.

En este dataset:

- **Company**: cadena de comida rápida (por ejemplo, McDonald's).
- **Item**: nombre del producto.
- **Calories**: energía total del producto (kcal).
- **Calories from Fat**: energía procedente de las grasas (kcal).
- **Total Fat (g)**: cantidad total de grasa.
- **Saturated Fat (g)**: grasa saturada.
- **Trans Fat (g)**: grasa trans.
- **Cholesterol (mg)**: colesterol.
- **Sodium (mg)**: sodio (sal).
- Otras columnas recogen hidratos de carbono, azúcares, proteínas y vitaminas.

Cada **fila representa un alimento concreto**, y cada **columna un nutriente o característica nutricional**.

Este tipo de tabla es muy similar a la información que encontramos en un **etiquetado nutricional**, pero ampliada.

In [ ]:
# Número de productos y número de variables
df.shape

(1147, 14)

In [ ]:
# Información sobre tipos de datos y valores no nulos
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1147 entries, 0 to 1146
Data columns (total 14 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Company               1147 non-null   object
 1   Item                  1147 non-null   object
 2   Calories              1147 non-null   object
 3   Calories from
Fat     642 non-null    object
 4   Total Fat
(g)         1091 non-null   object
 5   Saturated Fat
(g)     1091 non-null   object
 6   Trans Fat
(g)         1091 non-null   object
 7   Cholesterol
(mg)      1147 non-null   object
 8   Sodium 
(mg)          1147 non-null   object
 9   Carbs
(g)             1091 non-null   object
 10  Fiber
(g)             1091 non-null   object
 11  Sugars
(g)            1147 non-null   object
 12  Protein
(g)           1091 non-null   object
 13  Weight Watchers
Pnts  887 non-null    object
dtypes: object(14)
memory usage: 125.6+ KB


## Antes de empezar el análisis: revisión de los datos

Antes de analizar los alimentos y sacar conclusiones nutricionales, es imprescindible
**revisar cómo están los datos**. Este paso evita errores y malas interpretaciones.

### 1. ¿Qué significa `int64`?

En la salida de `df.info()` aparece el tipo de dato de cada columna.

- **int64** indica que la columna contiene **números enteros**.
- En este dataset, se utiliza para valores como:
  - calorías,
  - gramos de grasa,
  - miligramos de sodio,
  - etc.

Que una columna sea `int64` significa que:
- los datos son numéricos,
- se pueden ordenar, comparar y calcular,
- y son adecuados para análisis cuantitativos.

---

### 2. ¿Qué es un valor nulo?

Un **valor nulo** es un dato que falta y se representa como:
- `NaN` (Not a Number).

Esto puede ocurrir porque:
- el alimento no contiene ese nutriente,
- el dato no se ha medido,
- o la información no está disponible.

Un valor nulo **no es lo mismo que un valor cero**:
- `0` significa que el nutriente no está presente.
- `NaN` significa que **no sabemos el valor**.

---

### 3. ¿Por qué es importante revisar los valores nulos?

Si no revisamos los valores nulos antes de analizar:

- Las medias pueden salir incorrectas.
- Las comparaciones entre alimentos pueden ser engañosas.
- Podemos sacar conclusiones nutricionales erróneas.

Por eso, antes de empezar el análisis debemos:
1. Saber **qué columnas tienen valores nulos**.
2. Decidir si esos valores se:
   - eliminan,
   - sustituyen,
   - o simplemente se tienen en cuenta al interpretar.

Este paso es equivalente, en nutrición, a **comprobar si una etiqueta está completa antes de usarla**.

In [ ]:
# Comprobar cuántos valores nulos hay en cada columna
df.isnull().sum()

Company                    0
Item                       0
Calories                   0
Calories from\nFat       505
Total Fat\n(g)            56
Saturated Fat\n(g)        56
Trans Fat\n(g)            56
Cholesterol\n(mg)          0
Sodium \n(mg)              0
Carbs\n(g)                56
Fiber\n(g)                56
Sugars\n(g)                0
Protein\n(g)              56
Weight Watchers\nPnts    260
dtype: int64

## Interpretación de los valores nulos

Tras revisar los valores nulos del dataset, observamos lo siguiente:

### 1. Columnas sin valores nulos
Las siguientes variables **no tienen valores nulos**, por lo que se pueden usar con total confianza:

- Company
- Item
- Calories
- Cholesterol (mg)
- Sodium (mg)
- Sugars (g)

Esto significa que **todos los productos** tienen información completa para estos nutrientes.

---

### 2. Columnas con algunos valores nulos
Otras variables presentan valores nulos en **56 productos**:

- Total Fat (g)
- Saturated Fat (g)
- Trans Fat (g)
- Carbs (g)
- Fiber (g)
- Protein (g)

Esto indica que, para una parte pequeña del dataset, **no se dispone de esa información nutricional**.

---

### 3. Columnas con muchos valores nulos
La variable:

- Weight Watchers Pnts

presenta **260 valores nulos**, lo que indica que esta información no está disponible para la mayoría de los productos.

Desde un punto de vista nutricional y educativo, **esta variable no es prioritaria**, por lo que no se utilizará en el análisis principal.

---

### 4. ¿Qué significa esto antes de analizar?

Antes de empezar el análisis nutricional debemos decidir:

- Qué variables **vamos a usar**.
- Qué variables **vamos a descartar**.
- Y en qué casos debemos **tener cuidado al interpretar medias o comparaciones**.

En este notebook:
- Usaremos preferentemente variables **sin valores nulos o con muy pocos**.
- Evitaremos sacar conclusiones generales a partir de variables con muchos datos faltantes.

Este paso es equivalente, en nutrición, a **comprobar que una etiqueta nutricional está completa antes de basar en ella una recomendación**.

## Clave pedagógica: pensar antes de analizar

Este paso es fundamental desde el punto de vista formativo.

El resultado del análisis de valores nulos nos permite comprender que:

- **No todos los datasets están “perfectos”**.  
  Los datos reales, como la información nutricional, suelen presentar ausencias o incompletitudes.

- **Un valor nulo no es un error**, sino una **falta de información**.  
  No significa que el nutriente sea cero, sino que **no se dispone del dato**.

- **El análisis serio no empieza calculando**, sino **decidiendo qué variables usar y cuáles no**.  
  Elegir mal las variables puede llevar a conclusiones nutricionales incorrectas.

Este razonamiento es exactamente el mismo que se aplica en la práctica profesional:
antes de recomendar, comparar o evaluar un alimento, **hay que valorar la calidad de la información disponible**.

👉 Esto es **pensamiento profesional aplicado a la nutrición**, no programación.

## Selección de las variables para el análisis

Una vez revisada la estructura del dataset y los valores nulos, el siguiente paso es
**decidir qué variables vamos a utilizar en el análisis nutricional**.

Esta decisión no es técnica, es **metodológica**.

Para este notebook buscamos:
- Variables relevantes desde el punto de vista nutricional.
- Variables fáciles de interpretar en FP.
- Variables con información suficiente (pocos o ningún valor nulo).

---

### Variables identificativas
Estas variables nos permiten saber **de qué alimento estamos hablando**:

- **Company**: cadena de comida rápida.
- **Item**: nombre del producto.

---

### Variables nutricionales principales (sin valores nulos)
Estas variables se usarán como base del análisis:

- **Calories**: energía total.
- **Sugars (g)**: azúcares.
- **Sodium (mg)**: sodio.
- **Cholesterol (mg)**: colesterol.

Estas variables están completas para todos los productos y permiten realizar
comparaciones claras y fiables.

---

### Variables nutricionales secundarias (uso con precaución)
Las siguientes variables son relevantes, pero presentan algunos valores nulos:

- **Total Fat (g)**
- **Saturated Fat (g)**
- **Protein (g)**
- **Carbs (g)**
- **Fiber (g)**

Se utilizarán de forma puntual, indicando siempre que **no todos los productos
disponen de esta información**.

---

### Variable descartada
- **Weight Watchers Pnts**

Esta variable no se utilizará, ya que no es prioritaria desde el punto de vista
nutricional y presenta muchos valores nulos.

Esta selección nos permitirá realizar un análisis **claro, coherente y didáctico**,
sin introducir ruido innecesario.

In [ ]:
# Selección de las variables que vamos a utilizar en el análisis
variables_seleccionadas = [
    "Company",
    "Item",
    "Calories",
    "Sugars\n(g)",
    "Sodium \n(mg)",
    "Cholesterol\n(mg)",
    "Total Fat\n(g)",
    "Saturated Fat\n(g)",
    "Protein\n(g)",
    "Carbs\n(g)",
    "Fiber\n(g)"
]

df_sel = df[variables_seleccionadas]

# Comprobación de la nueva tabla
df_sel.head()

Company                           Item Calories Sugars\n(g)  \
0  McDonald’s                      Hamburger      250           6   
1  McDonald’s                   Cheeseburger      300           6   
2  McDonald’s            Double Cheeseburger      440           7   
3  McDonald’s                       McDouble      390           7   
4  McDonald’s   Quarter Pounder® with Cheese      510           9   

  Sodium \n(mg) Cholesterol\n(mg) Total Fat\n(g) Saturated Fat\n(g)  \
0           520                25              9                3.5   
1           750                40             12                  6   
2          1150                80             23                 11   
3           920                65             19                  8   
4          1190                90             26                 12   

  Protein\n(g) Carbs\n(g) Fiber\n(g)  
0           12         31          2  
1           15         33          2  
2           25         34          2  
3           22         33          2  
4           29         40          3

## Comentario didáctico: el criterio antes que el cálculo

Este paso es clave para entender cómo se trabaja con datos en un contexto profesional.

A partir de esta selección de variables, el alumnado aprende que:

- **No siempre se trabaja con todas las columnas** de un dataset.  
  Tener más datos no significa necesariamente tener mejor información.

- **Seleccionar variables es una decisión profesional**, no automática.  
  Depende del objetivo del análisis y del tipo de conclusiones que se quieran obtener.

- **El análisis no empieza calculando medias o haciendo gráficos**,  
  sino **decidiendo qué información es relevante y cuál no**.

Este razonamiento es el mismo que se aplica en nutrición y control alimentario:
antes de evaluar o comparar alimentos, **hay que decidir qué datos son realmente útiles**.

## Análisis del aporte energético (Calories)

La energía es uno de los primeros aspectos que se analizan en nutrición.

En este dataset, la variable **Calories** indica la **energía total** que aporta cada
producto de comida rápida, expresada en **kilocalorías (kcal)**.

Analizar el aporte energético nos permite:
- Comparar productos entre sí.
- Identificar alimentos con **alta densidad energética**.
- Valorar si un alimento puede consumirse de forma habitual u ocasional.

> Recordemos que un aporte elevado de calorías no implica automáticamente que un
alimento sea “malo”, pero sí obliga a **analizar su frecuencia y contexto de consumo**.

In [ ]:
# Estadísticos básicos de la variable Calories
df_sel["Calories"].describe()

count     1147
unique     105
top          0
freq        83
Name: Calories, dtype: object

## Claves para comentar los resultados en clase

Al observar los estadísticos del aporte energético, es recomendable detenerse a comentar:

- **El valor mínimo**, que indica el alimento con menor aporte calórico.
- **El valor máximo**, que muestra el alimento más energético del dataset.
- **La media**, que permite tener una idea del aporte calórico “típico” de estos productos.
- **La dispersión de los datos**, que refleja si los valores son similares entre sí o muy diferentes.

Este análisis ayuda a comprender que **no basta con mirar un solo valor**, sino que es necesario
interpretar el conjunto de los datos para sacar conclusiones nutricionales adecuadas.

## Relación entre calorías y grasas

Una vez analizado el aporte energético total, es importante estudiar
**qué nutrientes contribuyen más a esas calorías**.

Las **grasas** aportan **9 kcal por gramo**, más del doble que:
- los hidratos de carbono (4 kcal/g),
- y las proteínas (4 kcal/g).

Por este motivo, los alimentos con alto contenido graso suelen presentar
un **aporte energético elevado**, incluso aunque la cantidad consumida no sea grande.

En este apartado analizaremos:
- la grasa total,
- la grasa saturada,
- y su relación con las calorías.

In [ ]:
# Estadísticos básicos de la grasa total
df_sel["Total Fat\n(g)"].describe()

count     1091
unique      73
top          0
freq       357
Name: Total Fat\n(g), dtype: object

## Interpretación del contenido en grasa total

Al analizar la grasa total podemos observar que:

- Existen productos con **muy bajo contenido graso**.
- Otros productos presentan **cantidades elevadas de grasa**, lo que influye
directamente en su aporte energético.
- La variabilidad en el contenido graso explica, en gran parte, las diferencias
en calorías entre productos.

Esto refuerza la idea de que **no solo importa cuántas calorías tiene un alimento,
sino de dónde proceden esas calorías**.

In [ ]:
# Productos con mayor contenido en grasa total
df_sel[["Company", "Item", "Total Fat\n(g)", "Calories"]].sort_values(
    by="Total Fat\n(g)", ascending=False
).head(10)

Company                 Item Total Fat\n(g) Calories
464  Burger King   Barq’s® Root Beer‡                        
489  Burger King            Sweet Tea                        
479  Burger King   Hi-C® Fruit Punch‡                        
474  Burger King       Fanta® Orange‡                        
469  Burger King        Cherry Coke®‡                        
459  Burger King         Dr. Pepper®‡                        
454  Burger King             Sprite®‡                        
449  Burger King          Diet Coke®‡                        
444  Burger King  Coca Cola classic®‡                        
493  Burger King      Unsweetened Tea

## Grasa saturada y salud

Dentro de las grasas, la **grasa saturada** tiene especial relevancia
desde el punto de vista de la salud cardiovascular.

Un consumo elevado y frecuente de grasas saturadas se asocia con:
- aumento del colesterol LDL,
- mayor riesgo cardiovascular.

Por ello, no solo es importante analizar la cantidad total de grasa,
sino **el tipo de grasa presente en el alimento**.

In [ ]:
# Productos con mayor contenido en grasa saturada
df_sel[["Company", "Item", "Saturated Fat\n(g)", "Total Fat\n(g)"]].sort_values(
    by="Saturated Fat\n(g)", ascending=False
).head(10)

Company                         Item Saturated Fat\n(g)  \
464  Burger King           Barq’s® Root Beer‡                      
444  Burger King          Coca Cola classic®‡                      
469  Burger King                Cherry Coke®‡                      
484  Burger King  MINUTE MAID® Light Lemonade                      
459  Burger King                 Dr. Pepper®‡                      
489  Burger King                    Sweet Tea                      
454  Burger King                     Sprite®‡                      
449  Burger King                  Diet Coke®‡                      
474  Burger King               Fanta® Orange‡                      
479  Burger King           Hi-C® Fruit Punch‡                      

    Total Fat\n(g)  
464                 
444                 
469                 
484                 
459                 
489                 
454                 
449                 
474                 
479

## Conclusión del análisis de grasas

El análisis de las grasas muestra que:

- Los productos con mayor aporte calórico suelen tener un **alto contenido en grasa**.
- La grasa saturada representa una parte importante de esa grasa total en muchos productos.
- Desde el punto de vista nutricional, estos alimentos deberían consumirse
de forma **ocasional** y con moderación.

En el siguiente apartado analizaremos el papel de los **azúcares**, especialmente
en bebidas y postres de comida rápida.

## Clave didáctica: interpretar las grasas más allá de los números

En este bloque de análisis, el alumnado aprende a:

- **Relacionar los nutrientes con el aporte energético**, comprendiendo por qué
los alimentos ricos en grasa suelen ser más calóricos.

- **Diferenciar entre cantidad y calidad de grasa**, distinguiendo la grasa total
de la grasa saturada y valorando su impacto sobre la salud.

- **Justificar recomendaciones nutricionales utilizando datos reales**, y no solo
opiniones o ideas generales.

Este enfoque permite desarrollar un razonamiento nutricional sólido, basado en
información objetiva, que es esencial tanto en la práctica profesional como en la evaluación.

## Análisis del contenido en azúcares (Sugars)

Los **azúcares** forman parte de los hidratos de carbono, pero tienen una relevancia
especial desde el punto de vista nutricional y de salud pública.

Un consumo elevado y frecuente de azúcares se asocia con:
- aumento del aporte calórico,
- mayor riesgo de sobrepeso y obesidad,
- caries dental,
- y desplazamiento de alimentos nutricionalmente más interesantes.

En este apartado analizaremos el contenido en azúcares de los productos de comida rápida
y su impacto dentro de la dieta.

In [ ]:
# Estadísticos básicos de la variable Sugars
df_sel["Sugars\n(g)"].describe()

count     1147
unique     122
top          0
freq       190
Name: Sugars\n(g), dtype: object

## Interpretación inicial del contenido en azúcares

A partir de los valores obtenidos se observa que:

- Existen productos con **contenido muy bajo o nulo de azúcares**.
- Otros productos presentan **cantidades muy elevadas**, especialmente bebidas y postres.
- La variabilidad en el contenido de azúcares es muy alta entre los distintos productos.

Esto indica que los azúcares son un nutriente clave para **diferenciar alimentos**
dentro de la comida rápida.

In [ ]:
# Productos con mayor contenido en azúcares
df_sel[["Company", "Item", "Sugars\n(g)", "Calories"]].sort_values(
    by="Sugars\n(g)", ascending=False
).head(10)

Company                         Item Sugars\n(g) Calories
484   Burger King  MINUTE MAID® Light Lemonade                     
464   Burger King           Barq’s® Root Beer‡                     
474   Burger King               Fanta® Orange‡                     
489   Burger King                    Sweet Tea                     
469   Burger King                Cherry Coke®‡                     
493   Burger King              Unsweetened Tea                     
1054    Taco Bell   FOUNTAIN BEVERAGES (30 OZ)                     
1053    Taco Bell                                                  
912     Taco Bell           Soft Taco Supreme®                     
459   Burger King                 Dr. Pepper®‡

## Azúcares y tipo de producto

Los alimentos con mayor contenido en azúcares suelen corresponder a:
- bebidas azucaradas,
- batidos,
- postres y productos dulces.

En muchos casos, estos productos aportan:
- **muchas calorías**,
- **poca saciedad**,
- y **escaso valor nutricional**.

Esto los convierte en alimentos cuyo consumo debería ser **ocasional**,
especialmente en población infantil y adolescente.

## Conclusión del análisis de azúcares

El análisis del contenido en azúcares muestra que:

- Algunos productos de comida rápida contienen **cantidades muy elevadas de azúcares**.
- Estos productos contribuyen de forma significativa al **exceso calórico**.
- Desde el punto de vista nutricional, su consumo debería ser **limitado y ocasional**.

En el siguiente apartado se analizará otro nutriente crítico en la comida rápida:
el **sodio**, especialmente relevante en relación con la sal y la salud cardiovascular.

## Valor didáctico: comprender el impacto de los azúcares

En este bloque de análisis, el alumnado aprende a:

- **Identificar alimentos problemáticos por su elevado contenido en azúcares**, especialmente
bebidas y productos dulces.

- **Relacionar el contenido en azúcares con el aporte calórico y el tipo de producto**,
entendiendo por qué ciertos alimentos aportan muchas calorías con poca saciedad.

- **Justificar recomendaciones nutricionales utilizando datos reales**, desarrollando
un razonamiento basado en evidencias y no en percepciones generales.

Este enfoque permite conectar el análisis de datos con la educación nutricional y
con las competencias profesionales propias del ciclo formativo.

## Análisis del contenido en sodio (Sodium)

El **sodio** es un mineral necesario para el organismo, pero su consumo excesivo
está relacionado con importantes problemas de salud pública.

Un aporte elevado de sodio se asocia con:
- aumento de la presión arterial,
- mayor riesgo de hipertensión,
- incremento del riesgo cardiovascular.

Los alimentos de comida rápida suelen contener **cantidades elevadas de sodio**,
debido al uso de sal, salsas, conservantes y alimentos procesados.

En este apartado analizaremos el contenido en sodio de los productos del dataset
y su posible impacto nutricional.

In [ ]:
# Estadísticos básicos de la variable Sodium
df_sel["Sodium \n(mg)"].describe()

count     1147
unique     214
top          0
freq        54
Name: Sodium \n(mg), dtype: object

## Interpretación inicial del contenido en sodio

A partir de los estadísticos obtenidos se observa que:

- Algunos productos presentan **valores moderados de sodio**.
- Otros productos alcanzan **cantidades muy elevadas**, que pueden representar
una parte importante del consumo diario recomendado.
- La variabilidad en el contenido de sodio entre productos es muy alta.

Esto indica que el sodio es un nutriente clave para **evaluar la calidad nutricional**
de la comida rápida.

In [ ]:
# Productos con mayor contenido en sodio
df_sel[["Company", "Item", "Sodium \n(mg)", "Calories"]].sort_values(
    by="Sodium \n(mg)", ascending=False
).head(10)

Company                         Item Sodium \n(mg) Calories
489  Burger King                    Sweet Tea                       
449  Burger King                  Diet Coke®‡                       
474  Burger King               Fanta® Orange‡                       
484  Burger King  MINUTE MAID® Light Lemonade                       
912    Taco Bell           Soft Taco Supreme®                       
493  Burger King              Unsweetened Tea                       
469  Burger King                Cherry Coke®‡                       
479  Burger King           Hi-C® Fruit Punch‡                       
459  Burger King                 Dr. Pepper®‡                       
454  Burger King                     Sprite®‡

## Sodio y consumo diario recomendado

La Organización Mundial de la Salud recomienda **limitar el consumo de sodio**,
ya que un exceso mantenido en el tiempo aumenta el riesgo de hipertensión.

Muchos de los productos analizados pueden aportar:
- una gran parte del sodio diario recomendado,
- o incluso superarlo en una sola ingesta.

Esto hace especialmente importante:
- moderar su consumo,
- y prestar atención a la información nutricional.

In [ ]:
# Productos con menor contenido en sodio
df_sel[["Company", "Item", "Sodium \n(mg)", "Calories"]].sort_values(
    by="Sodium \n(mg)", ascending=True
).head(10)

Company                               Item Sodium \n(mg) Calories
573      Wendy’s        All-Natural Lemonade, Large             0      380
504  Burger King    BK® Café Decaf Coffee- 16 fl oz             0        0
148   McDonald’s  Minute Maid® Orange Juice (Small)             0      150
43    McDonald’s                              Honey             0       50
147   McDonald’s                      Dasani® Water             0        0
974    Taco Bell                              Water             0        0
608      Wendy’s                            Hot Tea             0        0
973    Taco Bell                Regular Iced Coffee             0       10
972    Taco Bell                 Premium Hot Coffee             0       10
388  Burger King         Mott’s® Natural Applesauce             0       50

## Conclusión del análisis de sodio

El análisis del sodio muestra que:

- Muchos productos de comida rápida contienen **cantidades elevadas de sodio**.
- El consumo frecuente de estos alimentos puede contribuir a **excesos de sal** en la dieta.
- Desde el punto de vista nutricional y de salud pública, es recomendable:
  - limitar su consumo,
  - y priorizar alimentos con menor contenido en sodio.

Este nutriente es clave para valorar el impacto de la comida rápida
sobre la salud cardiovascular.

## Actividad final tipo examen

A partir del análisis realizado del dataset de comida rápida, responde razonadamente
a las siguientes cuestiones.

### Pregunta 1
Identifica **un producto con alto aporte energético** y explica:
- qué nutrientes contribuyen principalmente a ese aporte,
- y por qué su consumo debería ser moderado.

---

### Pregunta 2
Selecciona **un producto con elevado contenido en azúcares** y justifica:
- por qué puede considerarse nutricionalmente problemático,
- y en qué tipo de población debería limitarse especialmente su consumo.

---

### Pregunta 3
Elige **un producto con alto contenido en sodio** y explica:
- qué riesgos puede suponer un consumo frecuente,
- y qué recomendaciones darías para reducir la ingesta de sal en la dieta.

---

### Pregunta 4
Compara dos productos del dataset (por ejemplo, una bebida y un plato principal) y razona:
- cuál sería una opción más adecuada desde el punto de vista nutricional,
- teniendo en cuenta calorías, azúcares y sodio.

> Recuerda: no se evalúa solo la respuesta final, sino la **justificación basada en datos**.

## Criterios de corrección (orientativos)

En esta actividad se valorará especialmente:

- La **correcta interpretación de los datos** del dataset.
- La capacidad para **relacionar nutrientes con efectos sobre la salud**.
- El uso de **argumentos nutricionales claros y bien estructurados**.
- La coherencia entre los datos analizados y las conclusiones obtenidas.

Las respuestas deben estar redactadas con lenguaje técnico adecuado al nivel
de FP de Dietética y Nutrición.

## Conclusiones finales

A lo largo de este notebook se ha trabajado con datos reales de alimentos de comida rápida
para desarrollar un análisis nutricional completo.

De forma general, se puede concluir que:

- La comida rápida presenta una **gran variabilidad nutricional**, tanto en energía
como en composición.
- Muchos productos destacan por su **alto contenido en calorías, grasas, azúcares o sodio**.
- El consumo frecuente de estos alimentos puede contribuir a **desequilibrios nutricionales**
y a un mayor riesgo de problemas de salud.

Este trabajo demuestra la importancia de:
- saber interpretar datos nutricionales,
- analizar alimentos con criterio,
- y justificar recomendaciones dietéticas basadas en información objetiva.

Estas competencias son fundamentales tanto para la **práctica profesional**
como para la evaluación en el ciclo formativo de Dietética y Nutrición.

## Cierre didáctico: qué hemos aprendido

Con este notebook el alumnado ha aprendido a:

- Trabajar con **datos reales** de alimentos.
- Revisar la calidad de un dataset antes de analizar.
- Seleccionar variables de forma razonada.
- Interpretar calorías, grasas, azúcares y sodio.
- Justificar decisiones y recomendaciones nutricionales con datos.

Este enfoque integra **nutrición, pensamiento crítico y análisis de datos**,
competencias clave en la formación profesional.